notificationworkerlambda/src/main/scala/com/gu/notifications/worker/delivery/fcm/oktransport/OkGoogleHttpRequest.scala

```diff
@@ -0,0 +1,100 @@
+package com.gu.notifications.worker.delivery.fcm.oktransport
+
+import java.io.ByteArrayInputStream
+import java.util.concurrent.ConcurrentLinkedQueue
+
+import com.google.api.client.http.{LowLevelHttpRequest, LowLevelHttpResponse}
+import okhttp3.{Headers, MediaType, OkHttpClient, Request, RequestBody, Response}
+import okio.BufferedSink
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuffer
+
+
+class OkGoogleHttpRequest(okHttpClient: OkHttpClient, url: String, method: String) extends LowLevelHttpRequest {
+  //No idea whether FCM now or in the future will call addHeaders concurrently
+  private val headers = new ConcurrentLinkedQueue[(String, String)]()
```

Comment:
I think this is a premature optimisation. An array, list or Map would do just fine

Reply:
After our chat, probably fine to keep like that